src/main/java/edu/kit/provideq/toolbox/qubo/solvers/QiskitQuboSolver.java

```diff
@@ -48,16 +47,17 @@ public void solve(Problem<String> problem, Solution<String> solution,
         .addProblemFilePathToProcessCommand()
         .addSolutionFilePathToProcessCommand()
         .problemFileName("problem.lp")
-        .run(problem.type(), solution.getId(), problem.problemData());
+        .run(getProblemType(), solution.getId(), input);
 
     // Return if process failed
     if (!processResult.success()) {
```

Comment:
To make all these setups using process runners easier and to avoid code duplication, we could maybe add a utility function to Solution so it can apply a process result.

Then we'd just have:
```java
solution.apply(processResult);
return Mono.just(solution);
```

instead of
```java
if (!processResult.success()) {
  solution.setDebugData(processResult.output());
  solution.abort();
  return Mono.just(solution);
}

solution.setSolutionData(processResult.output());
solution.complete();
return Mono.just(solution);
```

Reply:
A utility function seems like a good idea since the main goal of our process runners is to support the user with building problem solvers. However, I will add it to `ProcessResult` instead of `Solution` since I don't want our data model to depend on the process runners

Reply:
Done in 3d1907d and applied it to all places that allow for its usage

Reply:
Great, thanks a lot!